base/glibc-compatibility/musl/getauxval.c

```diff
@@ -26,10 +27,15 @@ __attribute__((constructor)) static void __auxv_init()
     size_t secure_idx = __find_auxv(AT_SECURE);
     if (secure_idx != ((size_t) -1))
         __auxv_secure = __auxv[secure_idx];
+
+    __auxv_initialized = 1;
 }
 
 unsigned long getauxval(unsigned long type)
 {
+    if (!__auxv_initialized)
+        __auxv_init();
```

Comment:
```suggestion
    if (!__auxv_initialized && __environ)
        __auxv_init();
```

Reply:
And also I'd add a simple check to the beginning of `__auxv_init()` to not initialize twice

```
if (__auxv_initialized)
  return;
```

Reply:
I've found simpler option. It looks it works.